flocker/node/script.py

```diff
@@ -76,6 +80,46 @@ def _get_external_ip(host, port):
         sock.close()
 
 
+def configuration_from_options(options):
+    """
+    Extract configuration from provided options.
+
+    :param dict options: Dictionary containing command line options.
+
+    :return dict: Dictionary containing the desired configuration.
+    """
+    control_service_options_file = options[u'control-service-config']
+
+    try:
+        options_content = control_service_options_file.getContent()
+    except IOError:
+        raise ConfigurationError(
+            "Configuration file does not exist at '{}'.".format(
+                control_service_options_file.path))
+
+    options_yaml = yaml.safe_load(options_content)
+
+    configuration = {}
+
+    try:
```

Comment:
We're presumably going to be loading a whole bunch of options this way. So maybe a utility function for "lookup and fail with message if not found."